hdfs/src/main/scala/akka/stream/alpakka/hdfs/model.scala

```diff
@@ -144,6 +153,21 @@ object FilePathGenerator {
   def apply(f: (Long, Long) => String, temp: String = DefaultTempDirectory): FilePathGenerator =
     new FilePathGenerator {
       val tempDirectory: String = temp
+      val newPathForEachFile: Boolean = NewPathForEachFile
+      def apply(rotationCount: Long, timestamp: Long): Path = new Path(f(rotationCount, timestamp))
+    }
+
+  /**
+   * Scala API: creates [[FilePathGenerator]] to rotate output
+   *
+   * @param f    a function that takes rotation count and timestamp to return path of output
+   * @param pathForEachFile turns on/off a possibility to create a new path for each file
+   *                        which depends on rotation count and timestamp supplied by hdfs messages
+   */
+  def apply(f: (Long, Long) => String, pathForEachFile: Boolean): FilePathGenerator =
```

Comment:
I think we should also support the `temp` directory argument in the other constructor.  Maybe we could create a type that uses the builder pattern and encapsulates the various options.

i.e.

```
class FilePathGeneratorSettings private(val tempDirectory: String, newPathForEachFile: Boolean) {
  def withTempDirectory(tempDir: String): FilePathGeneratorSettings = ...
  def withNewPathForEachFile(newPathForEachFile: Boolean): FilePathGeneratorSettings = ...
  private def copy ...
}
```

Reply:
Maybe we could add the optional `timestampExtractor` function to this new type instead of it being separate.